src/context/currentUser.js

```diff
@@ -0,0 +1,71 @@
+import { createContext, useEffect, useState } from 'react'
+import useAuth from '../hooks/useAuth'
+import jwt_decode from 'jwt-decode'
+import { getUser } from '../service/apiClient'
+
+export const CurrentUserContext = createContext()
+
+export const CurrentUserProvider = ({ children }) => {
+    const { token } = useAuth()
+    const [currentUser, setCurrentUser] = useState({ empty: true })
+    const [userInitials, setUserInitals] = useState('')
+    const [userFullName, setUserFullName] = useState('')
+    const [userCohort, setUserCohort] = useState('')
+    const [userFirstNameAndInital, setUserFirstNameAndInital] = useState('')
+
+    useEffect(() => {
+        async function getUserFromToken() {
+            if (token) {
+                const { userId } = jwt_decode(token)
+
+                const userDetails = await getUser(userId)
+                userDetails.status === 'success'
+                    ? setCurrentUser({ ...userDetails.data.user })
+                    : setCurrentUser({ empty: true })
+                return
+            }
+            setCurrentUser({ empty: true })
+            return
+        }
+        getUserFromToken()
+    }, [token])
+
+    useEffect(() => {
+        if (!currentUser.empty) {
+            setUserInitals(
+                currentUser
+                    ? `${currentUser.firstName[0].toUpperCase()}${currentUser.lastName[0].toUpperCase()}`
+                    : ''
+            )
```

Comment:
Do we need to keep the user's initials in state? If the app has access to the user's first and last name, it's safer to just run the uppercase on the first letter of each name when needed, if it's in state then you have another point in the app which could trigger re-renders or become stale